### ISSUE TITLE
link-to-external inconsistency between routable/routeless engines

### ISSUE THREAD

**Issue by dknutsen** [ember-engines/ember-engines#601]
### Use case:
I have two engines, one routable and one routeless. Both are rendering a specific component from a shared addon. Said component has a `link-to-external`. I want this link to work the same in both engines (given that they are configured the same way).

### Expected result:
I would expect to be able to give both engines the same external route in their engine configs (e.g. `foo: 'root.bar.foo'`) and then be able to use `{{#link-to-external 'foo' arg1 arg2 arg3}}...` from both engines the exact same way.

### Actual result:
The link works fine in the routable engine. In the routeless engine however it complains that route `foo` does not exist. If I look at [this line](https://github.com/ember-engines/ember-engines/blob/master/addon/components/link-to-external-component.js#L11) while it's running it looks like it's because the link-to-external won't look at the engine route map unless it has a mount point (aka is a routable engine?). If I change the link from `{{#link-to-external 'foo' arg1 arg2 arg3}}...` to `{{#link-to-external 'root.bar.foo' arg1 arg2 arg3}}...` it works, but then it breaks in the routable engine because `The external route 'root.bar.foo' does not exist`.

This seems very inconsistent to me. I don't see why a `link-to-external` would check to see if the engine is routable before mapping the route, and it seems even weirder that the behavior is then that the routeless engine can link to whatever owning app route it wants. [This assert](https://github.com/ember-engines/ember-engines/blob/master/addon/components/link-to-component.js#L14) suggests that `link-to-external` is the right way to go in a routeless engine, but the external link mechanics don't match up with routable engines.

### TL;DR
It seems like a `link-to-external` within a routeless engine should behave the same way as a routable engine but it does not, it behaves as if it were a link-to that isn't in the engine at all (which I think breaks the separation of concerns that engines are intended to have).

**Comment by dknutsen** [comment]
To illustrate this a little more clearly this is the workaround we've ended up with:
```
    {{#if inRoutableEngine}}
      {{#link-to-external 'routey' sid eid rid}}
        here
      {{/link-to-external}}.
    {{else}}
      {{#link-to-external 'foo.bar.routey' sid eid rid}}
        here
      {{/link-to-external}}.
    {{/if}}
```

When in the routable engine the `link-to-external` respects the external routes that we give to the engine ("routey" mapping to app route "foo.bar.routey"), but in the routeless engine (given the same external routes) it doesn't look for them and we have to pass the fully qualified route from the mounting app ("foo.bar.routey")

**Comment by thec0keman** [comment]
Thanks for finding a solution!
It seems like this is a rather important hole in boundary around engines, since it is possible to `link-to` anywhere in the host without the explicit mapping.
